src/web_resources/HangarXPLOR.LoadPage.js

```diff
@@ -10,9 +10,16 @@ HangarXPLOR.LoadPage = function(pageNo)
   
   var url = '/account/pledges?page=' + pageNo;
   
-  if (pageNo == 1 && document.location.search == '?page=1')
+  if(pageNo == 1) {
+    // TODO should this be place somewhere else? - it just needs to be done before we process the items
+    // so that we have the billing data already in order to link to it  
+    HangarXPLOR.Billing.LoadData();
```

Comment:
If we're only linking to billing data - perhaps the lookup could be executed on-click rather than pre-loaded?

Might help with larger hangars/billing history

Reply:
Correct. The look up happens when you click the arrow button to open up each item. The look ups should also always be fairly fast unless the user has a lot of bills on one specific day.